.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,45 @@
+name: Unit Tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [8, 11]
+    env:
+      CLOUDSDK_CORE_DISABLE_USAGE_REPORTING: true
+      CLOUDSDK_CORE_DISABLE_PROMPTS: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - uses: google-github-actions/setup-gcloud@v0.2.0
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
```

Comment:
what does this block do?

Reply:
`hashFiles` computes an aggregated SHA of all the given files, and the final hash is used as a key to save and retrieve the cache (~/.m2). That is, as long as those glob files remain untouched, this action will retrieve the same cache.